apptests/test_generators.py

```diff
@@ -0,0 +1,326 @@
+import json
+import unittest
+from copy import deepcopy
+from pathlib import Path
+from unittest.mock import MagicMock, mock_open, patch
+
+import generators.youtube_generator as gen
+
+empty_response_details = {
+    "etag": "YIUPVpqNjppyCWOZfL-19bLb7uk",
+    "items": [],
+    "kind": "youtube#videoListResponse",
+    "pageInfo": {"resultsPerPage": 0, "totalResults": 0},
+}
+example_response = {
+    "kind": "youtube#searchListResponse",
+    "etag": "wPpqdJzsY8IQM6ZYDqXcwWLEiVM",
+    "nextPageToken": "CAEQAA",
+    "regionCode": "US",
+    "pageInfo": {"totalResults": 273, "resultsPerPage": 1},
+    "items": [
+        {
+            "kind": "youtube#searchResult",
+            "etag": "NBmdLBv-G2RxAuLNA7IPe_s4x7Q",
+            "id": {"kind": "youtube#video", "videoId": "YazZwd48ws0"},
+            "snippet": {
+                "publishedAt": "2016-12-21T11:09:18Z",
+                "channelId": "UCWlvkaA27BCrcYG6gA0K8UA",
+                "title": "We create your Bucket List Trips and Events | Discover the Baltics with SMS Frankfurt Group Travel",
+                "description": "Tailormade Event and Incentives | Discover the Baltics with SMS Frankfurt Group Travel - We create your Bucket List Trips and ...",
+                "thumbnails": {
+                    "default": {
+                        "url": "https://i.ytimg.com/vi/YazZwd48ws0/default.jpg",
+                        "width": 120,
+                        "height": 90,
+                    }
+                },
+                "channelTitle": "SMS Frankfurt Group Travel",
+                "publishTime": "2016-12-21T11:09:18Z",
+            },
+        }
+    ],
+}
+parsed_example = [
+    {
+        "videoId": "YazZwd48ws0",
+        "channelId": "UCWlvkaA27BCrcYG6gA0K8UA",
+        "title": "We create your Bucket List Trips and Events | Discover the Baltics with SMS Frankfurt Group Travel",
+        "channelTitle": "SMS Frankfurt Group Travel",
+        "publishTime": "2016-12-21T11:09:18Z",
+    }
+]
```

Comment:
If the results_parser function works well, I think we can use the function with example_response as input instead of creating a global variable.